src/RA_httpthread.cpp

```diff
@@ -380,15 +380,14 @@ BOOL RAWeb::DoBlockingImageUpload(UploadType nType, const std::string& sFilename
 //  Adds items to the httprequest queue
 void RAWeb::CreateThreadedHTTPRequest(RequestType nType, const PostArgs& PostData, const std::string& sData)
 {
-    auto* pObj = new RequestObject(nType, PostData, sData);
-
-    ra::services::ServiceLocator::GetMutable<ra::services::IThreadPool>().RunAsync([pObj]()
+    ra::services::ServiceLocator::GetMutable<ra::services::IThreadPool>().RunAsync([nType, PostData, sData]() mutable
     {
+        auto pObj = std::make_unique<RequestObject>(nType, PostData, sData);
         std::string sResponse;
         DoBlockingRequest(pObj->GetRequestType(), pObj->GetPostArgs(), sResponse);
 
         pObj->SetResponse(sResponse);
-        ms_LastHttpResults.PushItem(pObj);
+        ms_LastHttpResults.PushItem(std::move(pObj)); // pointer gets corrupted if released here
```

Comment:
Made a `unique_ptr `overload. Nothing leaks. We can't `delete` here since it'll cause heap corruption and it's confusing, this is more simple.

Reply:
Should be able to eliminate the other `PushItem` implementation. I think this is the only place it's called.